File: Data Structures & Algorithms/generate-parentheses/submission-0.cpp
class Solution {
public:

    vector<string> res;

    void backtrack(string& current, int n, int open, int close) {
        if (current.length() == 2 * n) {
            res.push_back(current);
            return;
        }

        // Thêm dấu '('
        if (open < n) {
            current.push_back('(');
            backtrack(current, n, open + 1, close);
            current.pop_back();
        }   

        //Thêm dấu ')'
        if (close < open) {
            current.push_back(')');
            backtrack(current, n, open, close + 1);
            current.pop_back();
        }
    }

    vector<string> generateParenthesis(int n) {
        string current = "";
        backtrack(current, n, 0, 0);
        return res;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res, resLen = "", 0
        for i in range(len(s)):
            for j in range(len(s) - 1, i - 1, -1):
                l, r = i, j
                while l < r and s[l] == s[r]:
                    l += 1
                    r -= 1
                if l >= r and resLen < j - i + 1:
                    res = s[i: j + 1]
                    resLen = j - i + 1
        
        return res